### PR TITLE
[YUNIKORN-2342] Skeleton of the new UI 

### PR DESCRIPTION
**Pull request by doupache**
### What is this PR for?
Queue UI v2 

截圖 2024-03-08 晚上9 41 56

### Demo 
https://www.loom.com/share/826767ad1ea748e99cb661f6b816077a?sid=a9f900b5-8436-4f8a-b245-6494bba24753

### What type of PR is it?
* [x] - Feature

### What is the Jira issue?
[YUNIKORN-2342](https://issues.apache.org/jira/browse/YUNIKORN-2342)
[YUNIKORN-2343](https://issues.apache.org/jira/browse/YUNIKORN-2343)
[YUNIKORN-2345](https://issues.apache.org/jira/browse/YUNIKORN-2345)
[YUNIKORN-2346](https://issues.apache.org/jira/browse/YUNIKORN-2346)
[YUNIKORN-2348](https://issues.apache.org/jira/browse/YUNIKORN-2348)

### CODE REVIEW

**Comment by doupache** [comment]
cc @craigcondit @wilfred-s

**Comment by wilfred-s** [comment]
With the move to pnpm changes are needed. I got this to compile with the [attached patch](https://github.com/apache/yunikorn-web/files/14658131/pnpm_update.patch) using pnpm so all looks good from that side. If you can make the change to resolve the conflicts we are ready to commit.

**Comment by doupache** [comment]
Thanks @wilfred-s 
I switched this branch to `pnpm` by running the following command

```
1. git merge master 
2. make clean 
3. pnpm install 
4. make start-dev
```
Everything worked smoothly. 
I think it is ready to go 🎉

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

File: src/app/components/queue-v2/queues-v2.component.ts [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    private spinner: NgxSpinnerService,
+    private router: Router
+  ) {}
+
+  ngOnInit() {
+    this.fetchSchedulerQueuesForPartition()
+  }
+
+  fetchSchedulerQueuesForPartition() {
+    let partitionName = 'defau';
```
Comment (chenyulin0719) [inline src/app/components/queue-v2/queues-v2.component.ts:58]: Should be 'default'.

File: src/app/components/queue-v2/queues-v2.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+  describe('fetchSchedulerQueuesForPartition', () => {
+    it('should call SchedulerService and NgxSpinnerService methods', () => {
+      const schedulerSpy = spyOn(schedulerService, 'fetchSchedulerQueues').and.callThrough();
+      const spinnerShowSpy = spyOn(spinnerService, 'show').and.callThrough();
+      const spinnerHideSpy = spyOn(spinnerService, 'hide').and.callThrough();
+
+      component.fetchSchedulerQueuesForPartition();
+
+      expect(schedulerSpy).toHaveBeenCalledWith('defau');
```
Comment (chenyulin0719) [inline src/app/components/queue-v2/queues-v2.component.spec.ts:78]: ditto

File: src/app/components/queue-v2/queues-v2.component.ts (hunk 1 above)
Comment (chenyulin0719) [reply src/app/components/queue-v2/queues-v2.component.ts:58]: image

**Review by chenyulin0719 — CHANGES_REQUESTED** [review]
1 change required.

**Comment by doupache** [comment]
Thank you, @chenyulin0719, for the review.
I've also tested the web UI with the actual API.  
Everything works!

Please take a look.

**Review by chenyulin0719 — APPROVED** [review]
+1 LGTM

**Comment by doupache** [comment]
Thanks everyone for the help !

**Pull request closed without merging**